lib/pf/condition_parser.pm

```diff
@@ -0,0 +1,169 @@
+package pf::condition_parser;
+
+=head1 NAME
+
+pf::condition_parser - parser for filter logic
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::condition_parser
+
+Parses the following BNF
+
+EXPR = OR || OR
+EXPR = OR
+OR   = FACT && FACT
+OR   = FACT
+FACT = '(' EXPR ')'
+FACT = ID
+ID   = /a-zA-Z0-9_/+
+
+=cut
+
+use strict;
+use warnings;
+
+use base qw(Exporter);
+
+BEGIN {
+    our @EXPORT_OK = qw(parse_condition_string);
+}
+
+
+=head2 parse_condition_string
+
+Parses a string to a structure for building filters and conditions
+
+    my $array = parse_condition_string('(a || b) && (c || d)');
+
+Where $array would be structure
+
+    $array = [
+              'AND',
+              [
+                'OR',
+                'a',
+                [
+                  'AND',
+                  'b',
+                  'd'
+                ]
+              ],
+              [
+                'OR',
+                'c',
+                'd'
+              ]
+            ];
+
+If an invalid string is passed then it will die
```

Comment:
Shouldn't it return undef instead ?

We discussed the fact that we should reduce dying and this seems a bit overkill here